Lib/configparser.py

```diff
@@ -687,7 +688,7 @@ def read(self, filenames, encoding=None):
 
         Return list of successfully read files.
         """
-        if isinstance(filenames, str):
+        if isinstance(filenames, (str, bytes, os.PathLike)):
```

Comment:
configparser module doesn't support passing `bytes` objects at the moment. Accepting `bytes(some_pathlib_object)` would be a new feature and I think it's out of scope for this issue.

For example:

```rst
>>> import configparser as cp
>>> from test import support
>>> f = support.findfile("cfgparser.1")
>>> c = cp.ConfigParser()
>>> c.read(f.encode())
[]
>>> # If we pass ``[f.encode()]`` it will work.
>>> # However this doesn't documented anywhere in the docs.
>>> c.read([f.encode()])
[b'/home/berker/projects/cpython/master/Lib/test/cfgparser.1']
```